### DISCUSSION TITLE
Agentless Configuration for Network Device Management

### DISCUSSION THREAD

**Discussion by dtstanford** [gravitational/teleport#9517] [Q&A]
Hi all,

I see in the [FAQ](https://goteleport.com/docs/faq/) that agentless configuration of Teleport is an option, but I am not seeing any documentation elsewhere on how this might be configured.

My goal would be to configure Teleport as a single-pane-of-glass interface for management of 10s to 100s of networking devices (Cisco, Juniper, etc.) for managed services clients, utilizing the web browser Xterm.js terminal. A simple embedded user/password per device is all that would be needed.

Is this possible with Teleport? I can see this being a wide use case for other MSPs.

Thanks in advance!

**Comment by Valien** [comment]
Hi @dtstanford - Agentless mode is mainly for SSH-related functionality (using OpenSSH or similar). I opened up an issue around networking devices and you can follow it here - https://github.com/gravitational/teleport/issues/7343 

For what you are trying to accomplish you could add the Xterm.js as a potential web application and use Teleport's application service to manage that. Might require some configuration I would imagine.

**Comment by klizhentas** [comment]
as @Valien mentioned, Teleport supports agentless config with OpenSSH, Kubernetes, Applications and Databases in a sense that you don't need to run agent per SSH server, Kubernetes cluster, App and Database. 

You can find configs of how to join a regular OpenSSH node to Teleport cluster here:

https://goteleport.com/docs/server-access/guides/openssh/

With applications, databases and K8s clusters, one Teleport instance can be configured as Proxy for all these services. You can check docs for corresponding protocols in our documentation and let me know if you have any questions

**Reply by dtstanford** [reply]
Thanks, @klizhentas, but the documentation you've linked appears to describe server sshd configurations, not something applicable to networking devices. Or am I misunderstanding your suggestion?

**Reply by klizhentas** [reply]
@dtstanford what protocols are you planning to access those networking devices? We have customers who set up their routers with Teleport for remote access. Those routers were running OpenSSH though. Are you looking for tcp or custom protocol access?

**Reply by dtstanford** [reply]
We would only be looking for SSH access to our network devices using standard user/pass authentication. When reviewing the documentation, I did not see example configurations for this. I also stood up a Teleport instance, but was unable to figure out how I could create a basic session in this manner.

Could you share an example SSH configuration from the router side? Or do you mean they has Linux hosts/appliances operating as routers that they customized the SSH configuration for?

**Reply by webvictim** [reply]
Teleport itself doesn't provide any ability to connect to SSH servers using usernames/passwords - it will only authenticate using short-lived certificates for security. Some router OSes will allow you to modify `/etc/ssh/sshd_config` directly which would allow you to configure them with certificates to be used by Teleport - that's what's detailed in our OpenSSH guides.

If your router OS doesn't support this method, the recommendation is generally to have users connect to a regular "bastion" host inside your network which runs the Teleport agent - this will record all the sessions started on it. Then when users are connected to that host, they can run `ssh user@ip-of-router` from the console on that bastion host and use the router username/password to authenticate. That way you'll get a full session recording of all the commands run on the router itself, and you can use Teleport's audit log to see which underlying Teleport user actually connected to the shared bastion host and logged into the router.
